---
 core/libsparse/Makefile |   34 ++++++++++++++++++++++++++++++++++
 1 file changed, 34 insertions(+)

--- /dev/null
+++ android-tools/core/libsparse/Makefile
@@ -0,0 +1,34 @@
+SRCS += backed_block.c
+SRCS += output_file.c
+SRCS += sparse.c
+SRCS += sparse_crc32.c
+SRCS += sparse_err.c
+SRCS += sparse_read.c
+
+SRCSI  += img2simg.c
+SRCSS  += simg2img.c
+SRCSSS += simg2simg.c
+
+CPPFLAGS+= -Iinclude
+CPPFLAGS+= -I../../../external/zlib
+
+LIBS+= -lc -lz
+
+OBJS   = $(SRCS:.c=.o)
+OBJSI  = $(SRCSI:.c=.o)
+OBJSS  = $(SRCSS:.c=.o)
+OBJSSS = $(SRCSSS:.c=.o)
+
+all: simg2img simg2simg img2simg
+
+img2simg: $(OBJS) $(OBJSI)
+	$(CC) -o $@ $(LDFLAGS) $(OBJS) $(OBJSI) $(LIBS)
+
+simg2img: $(OBJS) $(OBJSS)
+	$(CC) -o $@ $(LDFLAGS) $(OBJS) $(OBJSS) $(LIBS)
+
+simg2simg: $(OBJS) $(OBJSSS)
+	$(CC) -o $@ $(LDFLAGS) $(OBJS) $(OBJSS) $(LIBS)
+
+clean:
+	rm -rf $(OBJS)
