Subject: Add makefile to build fastboot without using Android build system
Origin: https://heiher.info/2227.html

---
 core/fastboot/makefile |   41 +++++++++++++++++++++++++++++++++++++++++
 1 file changed, 41 insertions(+)

--- /dev/null
+++ android-tools-4.1.1/core/fastboot/makefile
@@ -0,0 +1,41 @@
+SRCS+= bootimg.c
+SRCS+= engine.c
+SRCS+= fastboot.c
+SRCS+= protocol.c
+SRCS+= usb_linux.c
+SRCS+= util_linux.c
+
+VPATH+= ../libzipfile
+SRCS+= centraldir.c
+SRCS+= zipfile.c
+
+VPATH+= ../../extras/ext4_utils/
+SRCS+= make_ext4fs.c
+SRCS+= backed_block.c
+SRCS+= ext4_utils.c
+SRCS+= indirect.c
+SRCS+= allocate.c
+SRCS+= contents.c
+SRCS+= output_file.c
+SRCS+= uuid.c
+SRCS+= extent.c
+SRCS+= sparse_crc32.c
+SRCS+= wipe.c
+SRCS+= sha1.c
+
+CPPFLAGS+= -I.
+CPPFLAGS+= -I../include
+CPPFLAGS+= -I../mkbootimg
+CPPFLAGS+= -I../../extras/ext4_utils/
+
+LIBS+= -lz
+
+OBJS= $(SRCS:.c=.o)
+
+all: fastboot
+
+fastboot: $(OBJS)
+	$(CC) -o $@ $(LDFLAGS) $(OBJS) $(LIBS)
+
+clean:
+	rm -rf $(OBJS) fastboot
