diff --git a/build/core/main.mk b/build/core/main.mk
index 0a7d3bf..b174db1 100644
--- a/build/core/main.mk
+++ b/build/core/main.mk
@@ -114,37 +114,37 @@ endif
 
 
 # Check for the correct version of java
-java_version := $(shell java -version 2>&1 | head -n 1 | grep '[ "]1\.6[\. "$$]')
-ifeq ($(strip $(java_version)),)
-$(info ************************************************************)
-$(info You are attempting to build with the incorrect version)
-$(info of java.)
-$(info $(space))
-$(info Your version is: $(shell java -version 2>&1 | head -n 1).)
-$(info The correct version is: 1.6.)
-$(info $(space))
-$(info Please follow the machine setup instructions at)
-$(info $(space)$(space)$(space)$(space)http://source.android.com/source/download.html)
-$(info ************************************************************)
-$(error stop)
-endif
-
-# Check for the correct version of javac
-javac_version := $(shell javac -version 2>&1 | head -n 1 | grep '[ "]1\.6[\. "$$]')
-ifeq ($(strip $(javac_version)),)
-$(info ************************************************************)
-$(info You are attempting to build with the incorrect version)
-$(info of javac.)
-$(info $(space))
-$(info Your version is: $(shell javac -version 2>&1 | head -n 1).)
-$(info The correct version is: 1.6.)
-$(info $(space))
-$(info Please follow the machine setup instructions at)
-$(info $(space)$(space)$(space)$(space)http://source.android.com/source/download.html)
-$(info ************************************************************)
-$(error stop)
-endif
-
+# java_version := $(shell java -version 2>&1 | head -n 1 | grep '[ "]1\.6[\. "$$]')
+# ifeq ($(strip $(java_version)),)
+# $(info ************************************************************)
+# $(info You are attempting to build with the incorrect version)
+# $(info of java.)
+# $(info $(space))
+# $(info Your version is: $(shell java -version 2>&1 | head -n 1).)
+# $(info The correct version is: 1.6.)
+# $(info $(space))
+# $(info Please follow the machine setup instructions at)
+# $(info $(space)$(space)$(space)$(space)http://source.android.com/source/download.html)
+# $(info ************************************************************)
+# $(error stop)
+# endif
+# 
+# # Check for the correct version of javac
+# javac_version := $(shell javac -version 2>&1 | head -n 1 | grep '[ "]1\.6[\. "$$]')
+# ifeq ($(strip $(javac_version)),)
+# $(info ************************************************************)
+# $(info You are attempting to build with the incorrect version)
+# $(info of javac.)
+# $(info $(space))
+# $(info Your version is: $(shell javac -version 2>&1 | head -n 1).)
+# $(info The correct version is: 1.6.)
+# $(info $(space))
+# $(info Please follow the machine setup instructions at)
+# $(info $(space)$(space)$(space)$(space)http://source.android.com/source/download.html)
+# $(info ************************************************************)
+# $(error stop)
+# endif
+ 
 $(shell echo 'VERSIONS_CHECKED := $(VERSION_CHECK_SEQUENCE_NUMBER)' \
         > $(OUT_DIR)/versions_checked.mk)
 endif
diff --git a/target/product/sdk.mk b/target/product/sdk.mk
index 7a986bd..d4c1064 100644
--- a/build/target/product/sdk.mk
+++ b/build/target/product/sdk.mk
@@ -188,9 +188,9 @@ PRODUCT_LOCALES := \
 	zh_TW
 
 # include available languages for TTS in the system image
-include external/svox/pico/lang/PicoLangDeDeInSystem.mk
-include external/svox/pico/lang/PicoLangEnGBInSystem.mk
-include external/svox/pico/lang/PicoLangEnUsInSystem.mk
-include external/svox/pico/lang/PicoLangEsEsInSystem.mk
-include external/svox/pico/lang/PicoLangFrFrInSystem.mk
-include external/svox/pico/lang/PicoLangItItInSystem.mk
+# include external/svox/pico/lang/PicoLangDeDeInSystem.mk
+# include external/svox/pico/lang/PicoLangEnGBInSystem.mk
+# include external/svox/pico/lang/PicoLangEnUsInSystem.mk
+# include external/svox/pico/lang/PicoLangEsEsInSystem.mk
+# include external/svox/pico/lang/PicoLangFrFrInSystem.mk
+# include external/svox/pico/lang/PicoLangItItInSystem.mk
